gef-ui/server/controllers/companies-house/index.js

```diff
@@ -24,36 +24,37 @@ const validateCompaniesHouse = async (req, res) => {
   const { params, body } = req;
   const { regNumber } = body;
   const { applicationId } = params;
+  const companiesHouseErrors = [];
+  let companiesHouseDetails;
+
+  if (!regNumber) {
+    companiesHouseErrors.push({
+      errRef: 'regNumber',
+      errMsg: 'Enter a Companies House registration number',
+    });
+  }
 
   try {
     const { exporterId } = await api.getApplication(applicationId);
 
-    if (_isEmpty(regNumber)) {
-      const regNumberError = {
-        errRef: 'regNumber',
-        errMsg: 'Enter a Companies House registration number',
-      };
-
-      return res.render('partials/companies-house.njk', {
-        errors: validationErrorHandler(regNumberError),
-        regNumber,
-        applicationId,
-      });
+    if (companiesHouseErrors.length === 0) {
+      companiesHouseDetails = await api.getCompaniesHouseDetails(regNumber, exporterId);
     }
 
-    await api.getCompaniesHouseDetails(regNumber, exporterId);
+    if (companiesHouseDetails && companiesHouseDetails.status === 422) {
```

Comment:
Is it only status 422 that gets returned on an error from CH? Should we check for a 200 instead and assume anything else is an error?